pkgs/development/python-modules/ydata-profiling/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, fastparquet
+, htmlmin
+, imagehash
+, jinja2
+, matplotlib
+, multimethod
+, numpy
+, pandas
+, phik
+, pydantic
+, pyyaml
+, requests
+, scipy
+, seaborn
+, statsmodels
+, tqdm
+, typeguard
+, visions
+}:
+
+buildPythonPackage rec {
+  pname = "ydata-profiling";
+  version = "4.1.2";
+  disabled = pythonOlder "3.7";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "ydataai";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-q0qbNVFKHJ/mrwtr3yRpFlU2hSY3CNiTXI3FIqfMSII=";
+  };
+
+  postPatch = ''
+    substituteInPlace requirements.txt  \
+       --replace "numpy>=1.16.0,<1.24" "numpy"  \
+       --replace "matplotlib>=3.2, <3.7" "matplotlib"  \
+       --replace "scipy>=1.4.1, <1.10" "scipy"
+  '';
```

Comment:
Could you use pythonRelaxDepsHook?
See, [Using pythonRelaxDepsHook {#using-pythonrelaxdepshook}](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook)

Reply:
I have used this hook before, but for some reason neither `pythonRelaxDeps = [ ... ]` nor `pythonRelaxDeps = true` worked.  I've seen this happen when reviewing also so perhaps it's a bug.

Reply:
OK, the error message says the version of ydata-profiling is `0.0.dev0`.
```
error: builder for '/nix/store/dpjzvdca8wraf4qvymq374rv3q4qwxqv-python3.10-ydata-profiling-4.1.2.drv' failed with exit code 1;
       last 10 log lines:
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./ydata_profiling-0.0.dev0-py2.py3-none-any.whl
       > Requirement already satisfied: pandas!=1.4.0,<1.6,>1.1 in /nix/store/2ng0kilm0ijh1z545dl5lyjkvimlzgfh-python3.10-pandas-1.5.3/lib/python3.10/site-packages (from ydata-profiling==0.0.dev0) (1.5.3)
       > Requirement already satisfied: htmlmin==0.1.12 in /nix/store/k77r9w60mmj1i0jsjbxg92d967sxlc2z-python3.10-htmlmin-0.1.12/lib/python3.10/site-packages (from ydata-profiling==0.0.dev0) (0.1.12)
       > ERROR: Could not find a version that satisfies the requirement numpy<1.24,>=1.16.0 (from ydata-profiling) (from versions: none)
       > ERROR: No matching distribution found for numpy<1.24,>=1.16.0
```
Looking at setup.py, I noticed that the version is set to "0.0.dev0" if no VERSION file is found.
https://github.com/ydataai/ydata-profiling/blob/01bd572044f40e859439e7fdd34f5b9cb1d4ecaf/setup.py#L14-L17

I also found that this file is created from the Makefile.
https://github.com/ydataai/ydata-profiling/blob/01bd572044f40e859439e7fdd34f5b9cb1d4ecaf/Makefile#L30
So I was able to build successfully by adding the following phases.

```nix
preBuild = ''
  echo ${version} > VERSION
'';
```

Reply:
OK, the error message says the version of ydata-profiling is `0.0.dev0`.
```
error: builder for '/nix/store/dpjzvdca8wraf4qvymq374rv3q4qwxqv-python3.10-ydata-profiling-4.1.2.drv' failed with exit code 1;
       last 10 log lines:
       > Finished executing setuptoolsBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./ydata_profiling-0.0.dev0-py2.py3-none-any.whl
       > Requirement already satisfied: pandas!=1.4.0,<1.6,>1.1 in /nix/store/2ng0kilm0ijh1z545dl5lyjkvimlzgfh-python3.10-pandas-1.5.3/lib/python3.10/site-packages (from ydata-profiling==0.0.dev0) (1.5.3)
       > Requirement already satisfied: htmlmin==0.1.12 in /nix/store/k77r9w60mmj1i0jsjbxg92d967sxlc2z-python3.10-htmlmin-0.1.12/lib/python3.10/site-packages (from ydata-profiling==0.0.dev0) (0.1.12)
       > ERROR: Could not find a version that satisfies the requirement numpy<1.24,>=1.16.0 (from ydata-profiling) (from versions: none)
       > ERROR: No matching distribution found for numpy<1.24,>=1.16.0
```
Looking at setup.py, I noticed that the version is set to "0.0.dev0" if no VERSION file is found.
https://github.com/ydataai/ydata-profiling/blob/01bd572044f40e859439e7fdd34f5b9cb1d4ecaf/setup.py#L14-L17

I also found that this file is created from the Makefile.
https://github.com/ydataai/ydata-profiling/blob/01bd572044f40e859439e7fdd34f5b9cb1d4ecaf/Makefile#L30
So I was able to build successfully by adding the following phases.

```nix
preBuild = ''
  echo ${version} > VERSION
'';
```

Reply:
Nice, thanks.

Reply:
Added this `preBuild`.